apps/framework-cli/tests/python/simple.py

```diff
@@ -0,0 +1,26 @@
+from dataclasses import dataclass, field
+from enum import Enum
+from typing import List, Optional
+
+
+type Key[T: (str, int)] = T 
+
+
+# Might be a solid idea to use the field specifier parameters to rename errand fields?
+# https://typing.readthedocs.io/en/latest/spec/dataclasses.html#field-specifier-parameters
+
+class Status(Enum):
+    OK = "ok"
+    ERROR = "error"
+
+
+@dataclass 
+class MyModel:
+    name: Key[str]
+    age: int
+    flag: bool
+    status: Status
+    test_key: str = field(alias="test-key")
```

Comment:
```
simple.py:24: error: No overload variant of "field" matches argument type "str"  [call-overload]
simple.py:24: note: Possible overload variants:
simple.py:24: note:     def [_T] field(*, default: _T, init: bool = ..., repr: bool = ..., hash: Optional[bool] = ..., compare: bool = ..., metadata: Optional[Mapping[Any, Any]] = ...) -> _T
simple.py:24: note:     def [_T] field(*, default_factory: Callable[[], _T], init: bool = ..., repr: bool = ..., hash: Optional[bool] = ..., compare: bool = ..., metadata: Optional[Mapping[Any, Any]] = ...) -> _T
simple.py:24: note:     def field(*, init: bool = ..., repr: bool = ..., hash: Optional[bool] = ..., compare: bool = ..., metadata: Optional[Mapping[Any, Any]] = ...) -> Any
```

It looks like `alias` is not a thing yet?
https://github.com/python/cpython/issues/101192

Reply:
What python version are you using

Reply:
```
python3.12 -m venv venv
source venv/bin/activate
pip install mypy
mypy simple.py
```

mypy version is 1.10.0

Reply:
```
python --version
Python 3.12.3
```

```
python simple.py
Traceback (most recent call last):
  File "/Users/george/stuff/simple.py", line 18, in <module>
    class MyModel:
  File "/Users/george/stuff/simple.py", line 23, in MyModel
    test_key: str = field(alias="test-key")
                    ^^^^^^^^^^^^^^^^^^^^^^^
TypeError: field() got an unexpected keyword argument 'alias'
```